src/structure_dump.cpp

```diff
@@ -457,20 +507,22 @@ void structure::load_chunk_layout(const char *filename, boundary_region &br) {
     }
     new_gv.set_origin(new_origin);
     gvs.push_back(new_gv);
+    ids.push_back(i * count_processors() / num_chunks);
   }
 
-  load_chunk_layout(gvs, br);
+  load_chunk_layout(gvs, ids, br);
 
   delete[] origins;
   delete[] nums;
 }
 
-void structure::load_chunk_layout(const std::vector<grid_volume> &gvs, boundary_region &br) {
+void structure::load_chunk_layout(const std::vector<grid_volume> &gvs,
+                                  const std::vector<int> &ids,
+                                  boundary_region &br) {
   // Recreate the chunks with the new grid_volumes
   for (int i = 0; i < num_chunks; ++i) {
     if (chunks[i]->refcount-- <= 1) delete chunks[i];
-    int proc = i * count_processors() / num_chunks;
-    chunks[i] = new structure_chunk(gvs[i], v, Courant, proc);
+    chunks[i] = new structure_chunk(gvs[i], v, Courant, ids[i]);
```

Comment:
Change to `ids[i] % count_processors()` and then we can use it with any number of processes.